pkg/compare/container_test.go

```diff
@@ -0,0 +1,326 @@
+package compare
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestIsContainer(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected bool
+	}{
+		{"container://my-image:latest:/metadata.yaml", true},
+		{"file://local/path/to/file.yaml", false},
+		{"http://example.com/data.yaml", false},
+		{"randomstringwithoutprefix", false},
```

Comment:
We could consider adding other potential edge cases such as:
- `"container:/only-one-slash:tag:/metadata.yaml"`
- `"container/with/no/colon/metadata.yaml"`
- `"container://name:tag"`
- `"container://name"`
- `"container://"`
- `"container`
- `""`